### PR TITLE
Gfc 751/sm/labels being swapped

### PR DESCRIPTION
**Pull request by SeanMurray1**

### CODE REVIEW

File: app/uk/gov/hmrc/gform/validation/ValidationService.scala [hunk 1]
```diff
@@ -193,9 +196,12 @@ class ComponentsValidator(
...
 
-  private def validateDate(fieldValue: FormComponent, date: Date): ValidatedType[Unit] = {
+  private def validateDate(
+    fieldValue: FormComponent,
+    date: Date,
+    otherFieldValue: Option[FormComponent]): ValidatedType[Unit] = {
     val reqFieldValidResult = validateDateRequiredField(fieldValue)
-    val otherRulesValidResult = validateDateImpl(fieldValue, date)(data)
+    val otherRulesValidResult = validateDateImpl(fieldValue, date, otherFieldValue)(data)
     Monoid[ValidatedType[Unit]].combineAll(List(reqFieldValidResult, otherRulesValidResult))
```
Comment (PasGat) [inline app/uk/gov/hmrc/gform/validation/ValidationService.scala]: Maybe we could inline the two variables passed as argument as they are only used one, so you could remove brackets too saving four lines of code ;-)

File: app/uk/gov/hmrc/gform/validation/ValidationService.scala [hunk 2]
```diff
@@ -250,7 +271,7 @@ class ComponentsValidator(
     concreteDate: ConcreteDate,
     offset: OffsetDate,
     data: FormDataRecalculated): Validated[GformError, Unit] =
-    validateInputDate(fieldValue, fieldValue.id, fieldValue.errorMessage, data).andThen(
+    validateInputDate(fieldValue, fieldValue.id, fieldValue.errorMessage, data, None).andThen(
```
Comment (PasGat) [inline app/uk/gov/hmrc/gform/validation/ValidationService.scala]: We seem to be calling `validateInputDate` with None at the end quite a lot - maybe and only maybe we could consider a default argument (not ideal I would normally discourage) or an overloaded function or else

File: app/uk/gov/hmrc/gform/validation/ValidationService.scala (hunk 1 above)
Comment (SeanMurray1) [reply app/uk/gov/hmrc/gform/validation/ValidationService.scala]: changes were resolved

File: app/uk/gov/hmrc/gform/validation/ValidationService.scala (hunk 2 above)
Comment (SeanMurray1) [reply app/uk/gov/hmrc/gform/validation/ValidationService.scala]: changes were resolved

**Pull request closed without merging**